tests/app/data_model/test_questionnaire_store.py

```diff
@@ -112,3 +116,138 @@ def test_questionnaire_store_raises_when_writing_to_metadata(questionnaire_store
 
     with pytest.raises(TypeError):
         store.metadata["no"] = "writing"
+
+
+class TestQuestionnaireStoreWithSupplementaryData:
+    store: QuestionnaireStore
+
+    def assert_list_store_data(self, list_name: str, list_item_ids: list[str]):
+        """Helper function to check that ListStore contains the given list with matching list_item_ids"""
+        lists = [list_model.name for list_model in self.store.list_store]
+        assert list_name in lists
+        assert self.store.list_store[list_name].items == list_item_ids
+
+    def test_adding_new_supplementary_data(
+        self, questionnaire_store, supplementary_data
+    ):
+        """Tests that adding supplementary data adds supplementary list items to the list store
+        this test doesn't mock list item ids, and checks that they match those in list_mappings
+        """
+        self.store = QuestionnaireStore(questionnaire_store.storage)
+        self.store.set_supplementary_data(supplementary_data)
+        assert "products" in self.store.supplementary_data_store.list_mappings
+        supplementary_list_item_ids = list(
+            self.store.supplementary_data_store.list_mappings["products"].values()
+        )
+        # check list mapping ids match list store ids
+        self.assert_list_store_data("products", supplementary_list_item_ids)
+
+    def test_updating_supplementary_data(
```

Comment:
I can see there is a test for this but when I functionally test by launching a survey with our current repeating mock data -> make a copy of the response id -> update the mock data so that an additional list item is added -> relaunch the mock endpoint -> relaunch the survey with same response id -> I still only see two items in the /dump/debug window when I would expect to see three 😕

Reply:
Great spot - but this is actually what we want. If the sds dataset id has not changed, then it won't call the endpoint again for a data update. If you want to see 3 you'd need to adjust the mock data and endpoint to have a different sds dataset id

Reply:
Ah i see I had `Any new list items in the reloaded data needs adding to the store.` in my head from the trello card but I see it is only when the id has changed 👍